modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,345 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Arr, Fun } from '@ephox/katamari';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import * as Chai from 'chai';
+import { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+import type Editor from 'tinymce/core/api/Editor';
+import type { User } from 'tinymce/core/lookup/UserLookup';
+
+Chai.use(chaiAsPromised);
+
+const createMockUser = (id: string): User => ({
+  id,
+  name: 'Test User',
+  avatar: 'test-avatar.png',
+  description: 'Test Description'
+});
+
+describe('browser.tinymce.core.UserLookupTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    user_id: 'test-user-1',
+    fetch_users: (userIds: string[]): Promise<User[]> => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const validIds = Arr.filter(userIds, (id): id is string =>
+            typeof id === 'string' && id.length > 0
+          );
+
+          // Only create users for valid test IDs
+          const users = Arr.bind(validIds, (id) =>
+            id.startsWith('test-user-') ? [ createMockUser(id) ] : []
+          );
+
+          resolve(users);
+        }, 0);
+      });
+    },
+  });
+
+  it('TINY-11974: Should return the configured user ID', () => {
+    const editor = hook.editor();
+    const currentUserId = editor.userLookup.getUserId();
+    expect(currentUserId).to.equal('test-user-1', 'Should return the configured user ID');
+  });
+
+  it('TINY-11974: Should have frozen getUserId but mutable fetchUsers', () => {
```

Comment:
You don't check `fetchUsers` at all. But I don't think you need to.

I don't think freezing a function makes a lot of sense. I expected this to just be a string, since the value is required on editor startup, and _then_ it makes sense to freeze it.